AdaptiveUiCodelab/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -291,3 +315,17 @@ fun ReplySearchBar(modifier: Modifier = Modifier) {
         )
     }
 }
+
+private fun PaddingValues.copy(
+    layoutDirection: LayoutDirection,
+    start: Dp? = null,
+    top: Dp? = null,
+    end: Dp? = null,
+    bottom: Dp? = null,
+) = PaddingValues(
+    start = start ?: calculateStartPadding(layoutDirection),
```

Comment:
I think this can be removed entirely with the alternate APIs above, but one subtle behavior issue that can arise with this code:

This causes `calculateStartPadding` and other `calculate*` methods to be invoked during composition. The correct inset values (as exposed in the `PaddingValues`) won't [necessarily be correct](https://developer.android.com/develop/ui/compose/layouts/insets#insets-and-phases) during composition, so anything based on the values that are computed during composition can end up resulting in being off by one frame.

A solution to this here would be to setup a `PaddingValues` that defers calling the `calculate*` values on the source `PaddingValues`, so that they are read later when the inset values will be correct:

```kotlin
fun PaddingValues.copy(
    start: Dp? = null,
    top: Dp? = null,
    end: Dp? = null,
    bottom: Dp? = null,
) = object : PaddingValues {
    override fun calculateBottomPadding(): Dp =
        bottom ?: this@copy.calculateBottomPadding()

    override fun calculateLeftPadding(layoutDirection: LayoutDirection): Dp =
        when (layoutDirection) {
            LayoutDirection.Ltr -> start
            LayoutDirection.Rtl -> end
        } ?: this@copy.calculateLeftPadding(layoutDirection)

    override fun calculateRightPadding(layoutDirection: LayoutDirection): Dp =
        when (layoutDirection) {
            LayoutDirection.Ltr -> end
            LayoutDirection.Rtl -> start
        } ?: this@copy.calculateRightPadding(layoutDirection)

    override fun calculateTopPadding(): Dp =
        top ?: this@copy.calculateTopPadding()
}
```

Reply:
acknowledged, thanks for the explanation!